proto/authn/v1/authn.proto

```diff
@@ -0,0 +1,71 @@
+// Authn protocol
+syntax = "proto3";
+package xmtp.authn.v1;
+
+import "google/api/annotations.proto";
+
+option go_package = "./authn/v1";
+
+// ----------------------------------------------------------------------------
+// These objects are copied from xmtpjs.
+// Compile failure occurs with GoGo protobuf as optional fields are not 
+// supported.
+// Migrating to go
+// ----------------------------------------------------------------------------
+
+// Signature represents a generalized public key signature,
+// defined as a union to support cryptographic algorithm agility.
+message Signature {
+    message ECDSACompact {
+        bytes bytes = 1; // compact representation [ R || S ], 64 bytes
+        uint32 recovery = 2; // recovery bit
+    }
+    oneof union {
+        ECDSACompact ecdsa_compact = 1;
+    }
+}
+
+message Secp256k1Uncompresed {
+    // uncompressed point with prefix (0x04) [ P || X || Y ], 65 bytes
+    bytes bytes = 1; 
+}
+// PublicKey represents a generalized public key,
+// defined as a union to support cryptographic algorithm agility.
+message PublicKey {
+    uint64 timestamp = 1;
+    Signature signature = 2;
+    oneof union {
+        Secp256k1Uncompresed secp256k1_uncompressed = 3;
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+message ClientAuthRequest {
+    PublicKey identity_key = 1;
+    bytes auth_data_bytes = 3;
+    Signature auth_data_signature = 4;
+}
+  
+message ClientAuthResponse {
+    oneof union {
+        bytes token = 1;  // if auth was successful
+        string error_str = 2; // if auth was not successful
+    }
+}
+  
+message AuthData {
+    string wallet_addr = 1;
+    uint64 timestamp = 3;
+}
+
+////////////////////////////
+
+service AuthnApi {
+    rpc Authenticate (ClientAuthRequest) returns (ClientAuthResponse) {
+        option (google.api.http) = {
+        post: "/authn/v1/authenticate"
```

Comment:
Feels like this and the next two lines should be indented. Does the formatter let you?

Reply:
I just run `dev/lint` on this and it mostly autofixes things. I'll check.